From 96b5306f9f2b5fe001fcc6b449d6ffc9dbe85461 Mon Sep 17 00:00:00 2001
From: Dmitry Lavnikevich <dmitry.lavnikevich@edality.by>
Date: Tue, 10 Apr 2018 19:59:18 +0300
Subject: [PATCH] fsl_ppfe: Fix phy autonegotiation

With 0x1a1 phy negotiation doesn't complete on the 0th phy. Was tested
on the LS1012a-based grapeboard.
---
 drivers/staging/fsl_ppfe/pfe_eth.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/staging/fsl_ppfe/pfe_eth.c b/drivers/staging/fsl_ppfe/pfe_eth.c
index 48377bd45582..11fd8f64d9fa 100644
--- a/drivers/staging/fsl_ppfe/pfe_eth.c
+++ b/drivers/staging/fsl_ppfe/pfe_eth.c
@@ -1486,7 +1486,7 @@ static void ls1012a_configure_serdes(struct net_device *dev)
 		pfe_eth_mdio_write(bus, 0, 0x13, 0x7);
 	} else {
 		pfe_eth_mdio_write(bus, 0, 0x14, 0xb);
-		pfe_eth_mdio_write(bus, 0, 0x4, 0x1a1);
+		pfe_eth_mdio_write(bus, 0, 0x4, 0x4001);
 		pfe_eth_mdio_write(bus, 0, 0x12, 0x400);
 		pfe_eth_mdio_write(bus, 0, 0x13, 0x0);
 	}
-- 
2.16.2

